text/0202-events-and-logs-api.md

```diff
@@ -0,0 +1,193 @@
+
+The effect of associating a Schema URL with a Logger MUST be that the telemetry emitted using the Logger will be associated with the Schema URL, provided that the emitted data format is capable of representing such association.
+
+### Logger
+
+The Logger is responsible for creating Events and Logs.
+
+Note that Loggers should usually not be responsible for configuration. This should be the responsibility of the LoggerProvider instead.
+
+#### Logger operations
+
+The Logger MUST provide functions to:
+
+- Function named “logEvent” to create an Event with the provided event name and attributes. The event name provided should be inserted as an attribute with key “event.name”. It will override any attribute with the same key in the attributes passed.
+- Function named “recordException” to record an Exception as an Event. This is to facilitate recording an exception outside of a trace context for languages that already do not support recording an exception in a log message. This should work similar to [Record Exception](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#record-exception) in Trace API, with the following semantics
+  - This is a specialized variant of logEvent function, so for anything not specified here, the same requirements as for logEvent apply.
+  - The signature of the method is to be determined by each language and can be overloaded as appropriate. The method MUST record an exception as an Event with the conventions outlined in the exception semantic conventions document. The minimum required argument SHOULD be no more than only an exception object.
+  - If RecordException is provided, the method MUST accept an optional parameter to provide any additional event attributes (this SHOULD be done in the same way as for the AddEvent method). If attributes with the same name would be generated by the method already, the additional attributes take precedence.
+  - Note: RecordException may be seen as a variant of logEvent with additional exception-specific parameters and all other parameters being optional (because they have defaults from the [exception semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md)).
+
+A LogRecord representing exception event may look like this:
+
+```
+{
+  time_unix_nano: 1653682410,
+  Attributes: {
+   event.name: exception,
+   exception.type:  OSError,
+   exception.message: Division by Zero,
+   exception.stacktrace: "",
+   exception.escaped: false
+ }
+}
+```
+
+The Logger SHOULD additionally provide the following functions:
+
+- Create a new Event using `LogRecord` data model.
```

Comment:
Why? Let's keep the API surface area as small as possible to start. I see the need for only two methods:
- logEvent
- logRecord

In java, we'd likely go the route of having a builder pattern for each of these.

Reply:
I think there is a misunderstanding here.  For logEvent, I meant methods like [these](https://github.com/scheler/opentelemetry-java/pull/1/files#diff-52068c2656fad153f3cd91781ec5028b30f400a258964282fb9b3b967a82f840R28). I would like to use the builder pattern only as a last option when I cannot create the Event using other methods.

Reply:
I think its unnecessary to specify the existence of builder methods and / or overloads which accept a variety of arguments. Those types of decision should be left to the languages. The spec should simply indicate that there is a method for logging events, that the method requires a name argument, and optionally accepts a variety of other arguments in line with the fields on LogRecord.

Now speaking as a maintainer for java, I think we'll likely choose to include a single API (either `logEvent(Event)` or the builder pattern `logEvent(String).setAttributes(..).build()`), but not multiple forms of the same API for convenience.

Reply:
If there was an API such as `logEvent(Event)`, then I think there would also have to be a way to create an instance of an `Event` through the API. I originally thought that this would have to be the case following the example of the Log SDK, which has `emit(LogRecord)` method.

I think we either need to keep this statement there, or the API should be just `logEvent(name, attributes)`.